### PR TITLE
chore: Address SonarCloud issues in xml-schema-ts

### PR DESCRIPTION
**Pull request by igarashitm**
https://github.com/KaotoIO/kaoto/issues/3008

 - Fixed two bugs in XmlSchemaDerivationMethod.schemaValueOf: the split regex used a literal \s (never matched spaces), and for...in iterated array indices instead of values
 - Reduced cognitive complexity in SchemaBuilder, XmlSchema, and XmlSchemaDerivationMethod by extracting focused helper methods (applyToken, findByQName, applyComplexTypeAttributes, handleElementConstraints, collectExtensionAttributes) and adding early-return guards
 - Replaced the 6 nearly-identical getXxxByQName methods in XmlSchema with a single generic findByQName<T> helper
 - Added readonly to fields that are initialized once and never reassigned across SchemaBuilder, XmlSchema, XmlSchemaCollection, and SchemaKey
 - Replaced == null / != null with strict === null / !== null (or ?? / ?? =) throughout the module and subdirectories
 - Added unit tests for XmlSchemaDerivationMethod edge cases and integration tests covering final, block, and abstract derivation method parsing through real XSD fixtures

## Release Notes

* **New Features**
  * Added support for parsing XML schema derivation methods, including final, block, and abstract attributes on complex types.

* **Improvements**
  * Enhanced internal code robustness through immutable field declarations and stricter null-checking patterns.
  * Consolidated schema resolution and element constraint handling to reduce code duplication.

### CODE REVIEW

File: packages/ui/src/xml-schema-ts/SchemaBuilder.ts [hunk 1]
```diff
@@ -863,22 +789,66 @@ export class SchemaBuilder {
...
+    while (keyEl !== null) {
+      element.getConstraints().push(this.handleConstraint(keyEl, new XmlSchemaKey()));
+      keyEl = XDOMUtil.getNextSiblingElementNS(keyEl, XmlSchema.SCHEMA_NS, 'key');
+    }
+    let keyrefEl = XDOMUtil.getFirstChildElementNS(el, XmlSchema.SCHEMA_NS, 'keyref');
+    while (keyrefEl !== null) {
+      const keyRef = this.handleConstraint(keyrefEl, new XmlSchemaKeyref()) as XmlSchemaKeyref;
+      if (keyrefEl.hasAttribute('refer')) {
+        keyRef.refer = this.getRefQName(keyrefEl.getAttribute('refer')!, el);
+      }
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/SchemaBuilder.ts]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/utils/XDOMUtil.ts [hunk 2]
```diff
@@ -69,11 +69,11 @@ export class XDOMUtil {
   static getNextSiblingElementNS(node: Node, uri: string, localpart?: string) {
     // search for node
     let sibling = node.nextSibling;
-    while (sibling != null) {
+    while (sibling !== null) {
       if (sibling.nodeType == Node.ELEMENT_NODE) {
         const siblingElement = sibling as Element;
         const siblingURI = siblingElement.namespaceURI;
-        if (siblingURI != null && siblingURI === uri && (localpart == null || siblingElement.localName === localpart)) {
+        if (siblingURI !== null && siblingURI === uri && (localpart === undefined || siblingElement.localName === localpart)) {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/utils/XDOMUtil.ts]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/XmlSchema.ts [hunk 3]
```diff
@@ -714,6 +543,34 @@ public void write(Writer writer, Map<String, String> options) {
...
+    const item = map.get(name) ?? null;
+    if (!deep || item !== null) {
+      return item;
+    }
+    for (const ext of this.externals) {
+      const schema = this.getSchema(ext);
+      if (schema !== null) {
+        const stack = schemaStack ?? [];
+        stack.push(this);
+        const found = deepGetter(schema, name, deep, stack);
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/XmlSchema.ts]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/extensions/ExtensionRegistry.ts [hunk 4]
```diff
@@ -9,8 +9,8 @@ export class ExtensionRegistry {
   /**
    * Maps for the storage of extension serializers /deserializers
    */
-  private extensionSerializers = new Map<string, ExtensionSerializer>();
-  private extensionDeserializers = new Map<QName, ExtensionDeserializer>();
+  private readonly extensionSerializers = new Map<string, ExtensionSerializer>();
+  private readonly extensionDeserializers = new Map<QName, ExtensionDeserializer>();
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/extensions/ExtensionRegistry.ts:13]: @coderabbitai ExtensionDeserializer is not in use and there's no plan to use it

File: packages/ui/src/xml-schema-ts/SchemaBuilder.ts [hunk 5]
```diff
@@ -649,14 +587,8 @@ export class SchemaBuilder {
...
-
-    if (this.resolvedSchemas != null) {
-      // cache is initialized, use it
-      const schemaKey = targetNamespace + schemaLocation + baseUri;
-      resolvedSchema = this.resolvedSchemas.get(schemaKey) || null;
-    }
-    return resolvedSchema;
+    const schemaKey = targetNamespace + schemaLocation + baseUri;
+    return this.resolvedSchemas.get(schemaKey) ?? null;
   }
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/SchemaBuilder.ts:592]: @coderabbitai it's not practically possible to cause a collision with this combination

File: packages/ui/src/xml-schema-ts/XmlSchema.ts [hunk 6]
```diff
@@ -46,7 +46,7 @@ export class XmlSchema extends XmlSchemaAnnotated implements NamespaceContextOwn
   constructor(namespace?: string, systemId?: string, parent?: XmlSchemaCollection) {
     super();
     if (namespace == null) return;
-    const systemIdToUse = systemId ? systemId : namespace;
+    const systemIdToUse = systemId || namespace;
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/XmlSchema.ts]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/attribute/XmlSchemaAttribute.ts [hunk 7]
```diff
@@ -81,7 +81,7 @@ export class XmlSchemaAttribute
   }
 
   setUse(use: XmlSchemaUse) {
-    if (this.namedDelegate.isTopLevel() && use != null) {
+    if (this.namedDelegate.isTopLevel() && use !== null) {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/attribute/XmlSchemaAttribute.ts]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/extensions/ExtensionRegistry.ts [hunk 8]
```diff
@@ -47,7 +47,7 @@ export class ExtensionRegistry {
   /**
    * Register a serializer with a Class
    *
-   * @param classOfType - the class of the object that would be serialized
+   * @param typeName
    * @param serializer - an instance of the deserializer
    */
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/extensions/ExtensionRegistry.ts:52]: @coderabbitai fixed

File: packages/ui/src/xml-schema-ts/utils/NodeNamespaceContext.ts [hunk 9]
```diff
@@ -16,14 +16,14 @@ export class NodeNamespaceContext implements NamespacePrefixList {
...
-    if (this.prefixes == null) {
+    if (this.prefixes === undefined) {
       this.prefixes = Object.keys(this.declarations);
     }
     return this.prefixes;
   }
 
   getNamespaceURI(pPrefix: string): string {
-    if (pPrefix == null) {
+    if (pPrefix === null) {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/utils/NodeNamespaceContext.ts]: @coderabbitai fixed
